docs/manual/src/proc_macro/index.md

```diff
@@ -120,13 +120,48 @@ pub trait Person {
 
 ## Conditional compilation
 
-`uniffi::constructor|method]` will work if wrapped with `cfg_attr` attribute:
+In many cases you can use Rust's `#[cfg()]` for conditional compilation.
+For example, with the `export` macro:
+
 ```rust
-#[cfg_attr(feature = "foo", uniffi::constructor)]
+#[cfg(target_os = "android")]
+#[uniffi::export]
+impl MyObject {
+    fn android_method(&self) -> String { ... }
+}
+
+#[cfg(target_os = "ios")]
+#[uniffi::export]
+impl MyObject {
+    fn ios_method(&self) -> String { ... }
+}
 ```
-Other attributes are not currently supported, see [#2000](https://github.com/mozilla/uniffi-rs/issues/2000) for more details.
+
+This works well when you want to gate the Rust implementation entirely.
+
+Note however that `#[cfg()]` does not work correctly inside `uniffi::export` blocks -
+uniffi scaffolding will always be generated even when not configured, resulting in compile errors.
```

Comment:
```suggestion
uniffi scaffolding will always be generated even when not configured, resulting in compile errors.
This can usually be worked around by using multiple `uniffi::export` blocks and placing the `#[cfg()]` attributes before `#[uniffi::export]`.
```